java-checks/src/main/java/org/sonar/java/checks/ConstantMathCheck.java

```diff
@@ -0,0 +1,195 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import com.google.common.collect.ImmutableList;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.methods.MethodInvocationMatcher;
+import org.sonar.java.checks.methods.MethodInvocationMatcherCollection;
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.ParenthesizedTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeCastTree;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import javax.annotation.CheckForNull;
+
+import java.util.List;
+
+@Rule(
+  key = "S2185",
+  name = "Silly math should not be performed",
+  tags = {"clumsy"},
+  priority = Priority.MAJOR)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.CPU_EFFICIENCY)
+@SqaleConstantRemediation("15min")
+public class ConstantMathCheck extends SubscriptionBaseVisitor implements JavaFileScanner {
+
+  private static final String CEIL = "ceil";
+  private static final String DOUBLE = "double";
+  private static final String FLOAT = "float";
+  private static final String FLOOR = "floor";
+  private static final String MATH_PACKAGE_NAME = "java.lang.Math";
+  private static final String ROUND = "round";
+
+  private static final MethodInvocationMatcherCollection TRUNCATION_METHODS = MethodInvocationMatcherCollection.create(
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(CEIL).addParameter(DOUBLE),
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(CEIL).addParameter(FLOAT),
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(FLOOR).addParameter(DOUBLE),
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(FLOOR).addParameter(FLOAT),
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(ROUND).addParameter(DOUBLE),
+    MethodInvocationMatcher.create().typeDefinition(MATH_PACKAGE_NAME).name(ROUND).addParameter(FLOAT)
+    );
```

Comment:
You forget to check about {{Math.rint}}. {{Math.abs}} is also mentioned in the RSPEC but not checked... However except if used explicitly with non-negative constant value, I'm not sure to see why it should be there.